src/node_errors.cc

```diff
@@ -820,9 +820,11 @@ void SetPrepareStackTraceCallback(const FunctionCallbackInfo<Value>& args) {
   env->set_prepare_stack_trace_callback(args[0].As<Function>());
 }
 
-static void EnableSourceMaps(const FunctionCallbackInfo<Value>& args) {
+static void SetSourceMapsEnabled(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  env->set_source_maps_enabled(true);
+  CHECK(args[0]->IsBoolean());
+  env->set_source_maps_enabled(
+      args[0].As<Boolean>()->BooleanValue(env->isolate()));
```

Comment:
`CHECK()` should only be used for things that are actually guaranteed – in this case, there is no typechecking – but the good part is that you also don’t need to verify the type here on the C++ side:

```suggestion
  env->set_source_maps_enabled(
      args[0]->BooleanValue(env->isolate()));
```

Reply:
Yeah, thanks for catching that. I forgot to add the checking on the JS side after experimenting with the features. Will do an update.

Reply:
@legendecas Even with the current change, you probably want

```suggestion
  CHECK(args[0]->IsBoolean());
  env->set_source_maps_enabled(
      args[0].As<Boolean>()->Value());
```

instead.

Reply:
@addaleax updated, PTAL again :D